File: API/DATA/ReceiptDetail.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DATA
{
    public partial class ReceiptDetail
    {
        public ReceiptDetail()
        {
            ReceiptImeis = new HashSet<ReceiptImei>();
        }

        public decimal ReceiptDetailID { get; set; }
        public decimal ReceiptID { get; set; }
        public string ProductID { get; set; } = null!;
        public string? CalculationUnit { get; set; }
        public decimal Quantity { get; set; }
        public decimal? UnitPrice { get; set; }
        public decimal? TotalAmount { get; set; }
        public decimal? TotalWeight1 { get; set; }
        public decimal? TotalWeight2 { get; set; }
        public decimal? TotalWeight3 { get; set; }
        public bool? IsImei { get; set; }
        public string? Description { get; set; }
        public DateTime? CreateDate { get; set; }
        public virtual Receipt Receipt { get; set; } = null!;
        public virtual ICollection<ReceiptImei> ReceiptImeis { get; set; }
    }
}

File: WEB/src/models/receiptDetail.ts
import receiptImei from './receiptImei';

interface receiptDetail {
    receiptDetailID: number;
    receiptID: number;
    productID: string; // mahh = productId
    calculationUnit: string;
    quantity: number;
    unitPrice: number;
    totalAmount: number;
    weight: number;
    totalWeight1: number;
    totalWeight2: number;
    totalWeight3: number;
    isImei: boolean;
    description: string;
    createDate?: Date,

}
export interface receiptDetailVm extends receiptDetail {
    productTypeID: string;
    productName: string;
    vuotton: boolean;
    index: number;
    sortOrder?: number;
    scaleNo: string;
    // employee: string,
    scaleEmployee?: string;
    scaleDate: Date;
    firstWeight?: number;
    secondWeight?: number;
    licensePlate?: string;
    ctimeis?: receiptImei[]
}
export default receiptDetail;